src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -860,6 +864,55 @@ private static void VerifyRefAssemblyClient(string lib_cs, string source, Action
         }
 
         [Theory]
+        [InlineData("")]
+        [InlineData(@"[assembly: System.Reflection.AssemblyVersion(""1"")]")]
+        [InlineData(@"[assembly: System.Reflection.AssemblyVersion(""1.0.*"")]")]
```

Comment:
It seems that the /deterministic flag affects two things: (1) whether wildcards are allowed in AssemblyVersion attribute, and (2) how PEs are emitted.
Would you know what the rationale is for (1)? It doesn't seem "non-deterministic".